src/resdk/collection_tables.py

```diff
@@ -0,0 +1,274 @@
+            slug=collection_slug
+        )  # type: Collection
+        if self.collection.samples.count() == 0:
+            raise ValueError(f"Collection {collection_slug} has no samples!")
+
+        self.cache_dir = cache_dir
+        if self.cache_dir is None:
+            self.cache_dir = default_cache_dir()
+        if not os.path.exists(self.cache_dir):
+            os.makedirs(self.cache_dir)
+
+        self._table_samples_ = []  # type: List[_TableSample]
+        self._meta = None  # type: Optional[pd.DataFrame]
+        self._expressions = {}  # type: Dict[str, pd.DataFrame]
+        self._gene_map_ = {}  # type: Dict[str, str]
+        self._modified = None  # type: Optional[datetime]
+
+    @property
+    def modified(self) -> datetime:
+        """Return the newest modification date out of all sample. Use to detect data changes on the server.
+
+        :return: modified datetime
+        """
+        if self._modified is None:
+            self._modified = self._get_modified()
+        return self._modified
+
+    def _get_modified(self) -> datetime:
+        """Get the newest modification date out of all sample in the collection."""
+        newest_modified = max(s.modified for s in self.collection.samples.iterate())
+        # transform into UTC so changing timezones won't effect cache
+        return newest_modified.astimezone(pytz.utc)
+
+    @lru_cache()
+    def metadata(self) -> pd.DataFrame:
+        """Return samples metadata in a table format.
+
+        :return: table of metadata
+        """
+        cache = load_cache(self.cache_dir, self.collection.slug, META, self.modified)
+        if cache is not None:
+            self._meta = cache
+
+        if self._meta is None:
+            self._meta = self._get_metadata()
+
+        save_cache(
+            self._meta, self.cache_dir, self.collection.slug, META, self.modified
+        )
+        return self._meta
+
+    def _get_metadata(self) -> pd.DataFrame:
+        """Flatten samples JSON metadata into a table."""
+        meta = pd.json_normalize([ts.metadata for ts in self._table_samples])
+        meta = meta.set_index("sample_slug").sort_index()
+        return meta
+
+    @property
+    def _table_samples(self) -> List[_TableSample]:
+        """Return a list of _TableSample objects containing expression files urls and metadata."""
+        if not self._table_samples_:
+            self._table_samples_ = self._get_table_samples()
+        return self._table_samples_
+
+    def _get_table_samples(self):
+        """Fetch expressions files urls and sample metadata."""
+        table_samples = []
+        for sample in self.collection.samples.iterate():
+            # get sample metadata from description
+            metadata = sample.descriptor
+            metadata["sample_slug"] = sample.slug
+
+            # get expressions data objects
+            try:
+                exp_data = sample.get_expression()
```

Comment:
Since this is an object returned by `get_expression()` method, `expression` would be a perfect name for it.